go/vt/vttablet/onlineddl/executor.go

```diff
@@ -1473,6 +1496,129 @@ func (e *Executor) executeRevert(ctx context.Context, onlineDDL *schema.OnlineDD
+		}()
+
+	}
+
+	// Compare the existing (to be potentially migrated) table with the declared (newly created) table:
+	// all things are tengo related
+	if err := func() error {
+		variables, err := e.readMySQLVariables(ctx)
+		if err != nil {
+			return err
+		}
+		flavor := tengo.ParseFlavor(variables.version, variables.versionComment)
+
+		// Create a temporary account for tengo to use
+		onlineDDLPassword, err := e.createOnlineDDLUser(ctx)
+		if err != nil {
+			return err
+		}
+		defer e.dropOnlineDDLUser(ctx)
+
+		// tengo requires sqlx.DB
+		cfg := mysqldriver.NewConfig()
+		cfg.User = onlineDDLUser
+		cfg.Passwd = onlineDDLPassword
+		cfg.Net = "tcp"
+		cfg.Addr = fmt.Sprintf("%s:%d", variables.host, variables.port)
+		cfg.DBName = e.dbName
+		cfg.ParseTime = true
+		cfg.InterpolateParams = true
+		cfg.Timeout = 1 * time.Second
+		mysqlDSN := cfg.FormatDSN()
+
+		db, err := sqlx.Open("mysql", mysqlDSN)
+		if err != nil {
+			return err
+		}
+		defer db.Close()
+
+		// Read existing table
+		existingTable, err := tengo.QuerySchemaTable(ctx, db, e.dbName, onlineDDL.Table, flavor)
+		if err != nil {
+			return err
+		}
+		// Read comparison table
+		comparisonTable, err := tengo.QuerySchemaTable(ctx, db, e.dbName, comparisonTableName, flavor)
+		if err != nil {
+			return err
+		}
+		// We created the comparison tablein same schema as original table, but under different name (because obviously we can't have
+		// two tables with identical name in same schema). It is our preference to create the table in the same schema.
+		// unfortunately, tengo does not allow comparing tables with different names. After asking tengo to read table info, we cheat
+		// and override the name of the table:
+		comparisonTable.Name = existingTable.Name
+		// We also override `.CreateStatement` (output of SHOW CREATE TABLE), because tengo has a validation, where if it doesn't
+		// find any ALTER changes, then the CreateStatement-s must be identical (or else it errors with UnsupportedDiffError)
+		comparisonTable.CreateStatement, err = schema.ReplaceTableNameInCreateTableStatement(comparisonTable.CreateStatement, existingTable.Name)
+		if err != nil {
+			return err
+		}
+		// Diff the two tables
+		diff := tengo.NewAlterTable(existingTable, comparisonTable)
+		if diff == nil {
+			// No change. alterClause remains empty
+			return nil
+		}
+		mods := tengo.StatementModifiers{
+			AllowUnsafe: true,
+			NextAutoInc: tengo.NextAutoIncIfIncreased,
+		}
+		alterClause, err = diff.Clauses(mods)
+		if err != nil {
+			return err
+		}
+		return nil
+	}(); err != nil {
+		return "", err
+	}
+
+	return alterClause, nil
+}
+
+// executeMigration executes a single migration. It analyzes the migration type:
+// - is it declarative?
+// - is it CREATE / DROP / ALTER?
+// - it is a Revert request?
+// - what's the migration strategy?
+// The function invokes th eappropriate handlers for each of those cases.
```

Comment:
th eappropriate handlers => the appropriate handlers

Reply:
fixed